mailhandler/spameater

```diff
@@ -813,7 +814,7 @@ sub sendMail {
   return if !$rcpt;
   $config->debug("sending...") if $extradebug;
   if ((!$useXHeader || $useXHeader==2 && $subjecttext =~ /message/) && $subjecttext) {
-    my $check = $$msgref =~ s/(^Subject\:.*$)/$1$subjecttext/mi;
+    my $check = $$msgref =~ s/(^Subject\:(.|(\n\s))*$)/$1$subjecttext/mi;
```

Comment:
Not sure what's the purpose of this regex. `(.|(\n\s))*` matches "Anything or a new line and space, repeated 0 or more". Unless I am mistaken, it allows the Subject field in the message to be split in several space-prefixed lines. Can you confirm my thinking?

Reply:
Yes, that's also what @mikedlr and I thought. And it seems right, because the subject line might be broken over multiple indented lines. (At least, @mikedlr seemed to recall that this is allowed. I don't stare enough at raw emails to have much of an opinion about that, honestly).

Reply:
Split across several space-prefixed lines, or also pairs of double new lines from the looks of it. Since `\s` can also match on `\n`.

The following were valid matches:
```
Subject:
```
```
subject: test 123
```
Now the following are also valid matches
```
Subject: first line
 second line
```
```
Subject: first line
 second line



last line
```

Reply:
Hmm, well spotted. Then this would also match:

```
Subject: first line

Date: 1234-56-78
```

Is that allowed?

Reply:
my belief is that https://www.rfc-editor.org/rfc/rfc2822#section-2.2.2 and the subsequent discussion on folding long mail lines says that double new line is not allowed. Folding whitespace should only be either space (ASCII 32) or TAB (ASCII 9). 

We should build testcases and check this.

Reply:
@mikedlr Could you do that? I'm not competent enough with perl (yet) to reliably test this.